src/crystal/dwarf/line_numbers.cr

```diff
@@ -212,25 +214,65 @@ module Crystal
           sequence.offset = offset
           sequence.unit_length = @io.read_bytes(UInt32)
           sequence.version = @io.read_bytes(UInt16)
+
+          if sequence.version < 2 || sequence.version > 5
+            # return Error.new("unknown line table version: #{version}")
+          end
+
+          if sequence.version >= 5
+            sequence.address_size = @io.read_bytes(UInt8).to_i
+            sequence.segment_selector_size = @io.read_bytes(UInt8).to_i
+          else
+            sequence.address_size = {{ flag?(:bits64) ? 8 : 4 }}
+            sequence.segment_selector_size = 0
+          end
+
           sequence.header_length = @io.read_bytes(UInt32)
-          sequence.minimum_instruction_length = @io.read_byte.not_nil!
+          sequence.minimum_instruction_length = @io.read_bytes(UInt8).to_i
 
           if sequence.version >= 4
-            sequence.maximum_operations_per_instruction = @io.read_byte.not_nil!
+            sequence.maximum_operations_per_instruction = @io.read_bytes(UInt8).to_i
+          else
+            sequence.maximum_operations_per_instruction = 1
           end
 
-          sequence.default_is_stmt = @io.read_byte.not_nil! == 1
-          sequence.line_base = @io.read_bytes(Int8)
-          sequence.line_range = @io.read_byte.not_nil!
-          sequence.opcode_base = @io.read_byte.not_nil!
+          if sequence.maximum_operations_per_instruction == 0
+            # return Error.new("invalid maximum operations per instruction: 0")
+          end
 
+          sequence.default_is_stmt = @io.read_byte == 1
+          sequence.line_base = @io.read_bytes(Int8).to_i
+          sequence.line_range = @io.read_bytes(UInt8).to_i
+          if sequence.line_range == 0
+            # return Error.new("invalid line range: 0")
+          end
+
+          sequence.opcode_base = @io.read_bytes(UInt8).to_i
           read_opcodes(sequence)
-          read_directory_table(sequence)
-          read_filename_table(sequence)
+
+          if sequence.version < 5
+            read_directory_table(sequence)
+            read_filename_table(sequence)
+          else
+            dir_format = read_lnct_format
+            count = DWARF.read_unsigned_leb128(@io)
+            count.times do
+              dir, _, _ = read_lnct(sequence, dir_format)
+              sequence.include_directories << dir
+            end
+
+            file_format = read_lnct_format
+            count = DWARF.read_unsigned_leb128(@io)
+            count.times do
+              name, mtime, size = read_lnct(sequence, file_format)
+              sequence.file_names << {name, mtime.to_i, size.to_i, 0}
+            end
```

Comment:
Maybe extract 2 explicit methods, similar to the ones for DWARF < 4? E.g. `read_lnct_directory_table` or something similar.